### PR TITLE
conformance: zone that lacks DS in parent zone

### PR DESCRIPTION
**Pull request by japaric**
port of `no-ds` from EDE dot com

like #2380 this is another "Insecure" scenario where hickory is answering with SERVFAIL instead of NOERROR+ad=false

### CODE REVIEW

File: conformance/packages/conformance-tests/src/resolver/dnssec/scenarios/insecure.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    nameservers_ns.add(nameservers_ns.a());
+
+    root_ns.referral_nameserver(&tld_ns);
+    tld_ns.referral_nameserver(&nameservers_ns);
+    tld_ns.referral_nameserver(&no_ds_ns);
+
+    let nameservers_ns = nameservers_ns.sign(sign_settings.clone())?;
+
+    tld_ns.add(nameservers_ns.ds().clone());
+    // tld_ns.add(no_ds_ns.ds().clone()); // IMPORTANT omit this!
```
Comment (bluejekyll) [inline conformance/packages/conformance-tests/src/resolver/dnssec/scenarios/insecure.rs:45]: Can you update the comment as to why?

**Review by bluejekyll — CHANGES_REQUESTED** [review]
Just want a little more info on a comment, otherwise looks fine to me.

**Comment by japaric** [comment]
updated to actually sign the `no_ds` zone and to not use variables like `nameservers_ns` (see #2408 )

**Comment by japaric** [comment]
@bluejekyll reminder that this PR cannot be merged until you update your review

**Review by marcus0x62 — APPROVED** [review]

**Comment by japaric** [comment]
> @bluejekyll reminder that this PR cannot be merged until you update your review

@bluejekyll ping ^

**Review by justahero — APPROVED** [review]

**Comment by japaric** [comment]
is `e2e_tests::tx_id_validation_test` known to spuriously fail? the test is not modified by this PR and this PR does not modify hickory-dns code either. I cannot repro the failure locally after running `just e2e-tests` dozens of times. cc @marcus0x62 

[from this build job](https://github.com/hickory-dns/hickory-dns/actions/runs/11577413801/job/32228737228?pr=2388#step:10:461):

``` console
failures:

---- recursor::security::scenarios::tx_id_validation_test stdout ----
STDOUT:
;; communications error to 192.168.160.4#53: timed out
;; communications error to 192.168.160.4#53: timed out
;; communications error to 192.168.160.4#53: timed out

; <<>> DiG 9.18.28-1~deb12u2-Debian <<>> +recurse +nodnssec +adflag +nocdflag @192.168.160.4 A www.example.testing.
; (1 server found)
;; global options: +cmd
;; no servers could be reached
STDERR:

thread 'recursor::security::scenarios::tx_id_validation_test' panicked at src/recursor/security/scenarios.rs:55:9:
error
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by marcus0x62** [comment]
> is `e2e_tests::tx_id_validation_test` known to spuriously fail? the test is not modified by this PR and this PR does not modify hickory-dns code either. I cannot repro the failure locally after running `just e2e-tests` dozens of times. cc @marcus0x62
> 
> [from this build job](https://github.com/hickory-dns/hickory-dns/actions/runs/11577413801/job/32228737228?pr=2388#step:10:461):
> 
> ```
> failures:
> 
> ---- recursor::security::scenarios::tx_id_validation_test stdout ----
> STDOUT:
> ;; communications error to 192.168.160.4#53: timed out
> ;; communications error to 192.168.160.4#53: timed out
> ;; communications error to 192.168.160.4#53: timed out
> 
> ; <<>> DiG 9.18.28-1~deb12u2-Debian <<>> +recurse +nodnssec +adflag +nocdflag @192.168.160.4 A www.example.testing.
> ; (1 server found)
> ;; global options: +cmd
> ;; no servers could be reached
> STDERR:
> 
> thread 'recursor::security::scenarios::tx_id_validation_test' panicked at src/recursor/security/scenarios.rs:55:9:
> error
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```

I haven't seen failures with that test so far, and while it does use some new features I added to the conformance suite (the leaf resolver in that test is a python script,) that error looks like a problem with the client being unable to communicate with the resolver and none of that should be unusual or unique...

**Pull request merged**